shell.nix

```diff
@@ -23,6 +23,25 @@ mkShell {
   ] ++ lib.optionals docTools [graphviz python36Packages.sphinx zip unzip];
 
   shellHook = ''
+    # Add nix config flags to .bazelrc.local.
+    #
+    BAZELRC_LOCAL=".bazelrc.local"
+    if [ ! -e "$BAZELRC_LOCAL" ] || (\
+       ! grep -s -q "\-\-host_platform\=" "$BAZELRC_LOCAL" && \
```

Comment:
We're invariably going to get false negatives here (aliases, implied flags via `--config` etc). I'd simplify the script to the point that we're only testing for the existence of `.bazelrc.local`. If it already exists, then presumably the user knows what they're doing.